src/ModelContextProtocol/Authentication/Types/ProtectedResourceMetadata.cs

```diff
@@ -0,0 +1,145 @@
+using System.Text.Json.Serialization;
+
+namespace ModelContextProtocol.Types.Authentication;
+
+/// <summary>
+/// Represents the resource metadata for OAuth authorization as defined in RFC 9396.
+/// Defined by <see href="https://datatracker.ietf.org/doc/rfc9728/">RFC 9728</see>. 
+/// </summary>
+public class ProtectedResourceMetadata
+{
+    /// <summary>
+    /// The resource URI.
+    /// </summary>
+    /// <remarks>
+    /// REQUIRED. The protected resource's resource identifier.
+    /// </remarks>
+    [JsonPropertyName("resource")]
+    public Uri Resource { get; set; } = null!;
```

Comment:
This null assertion can lead to bypassing of compile time checks, and exceptions at runtime. It would be safer to remove the setter and use "init" instead and combine with the "required" keyword.